### ISSUE TITLE
uninitialized constant Doorkeeper::Application while booting

### ISSUE THREAD

**Issue by masterkain** [doorkeeper-gem/doorkeeper#1583]
I have a rails 6.1 app, with factory_bot, doorkeeper and a factory such as

```
FactoryBot.define do
  factory :application, class: 'Doorkeeper::Application' do
    name { 'oauth client app' }
    redirect_uri { 'https://oauthclient.dev/users/auth/account/callback' }
  end
end
```

my doorkeeper config has `enable_application_owner confirmation: true`, I already tried to put `require 'doorkeeper/orm/active_record/application'` atop the file with the same result.

when I load the test environment through guard I get, even if the specified factory class is a string:

```
rake aborted!
NameError: uninitialized constant Doorkeeper::Application
Did you mean?  ApplicationJob
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/inflector/methods.rb:288:in `const_get'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/inflector/methods.rb:288:in `block in constantize'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/inflector/methods.rb:284:in `each'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/inflector/methods.rb:284:in `inject'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/inflector/methods.rb:284:in `constantize'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/core_ext/string/inflections.rb:74:in `constantize'
/bundle/vendor/ruby/3.0.0/gems/doorkeeper-5.5.4/lib/doorkeeper/config.rb:465:in `application_model'
/bundle/vendor/ruby/3.0.0/gems/doorkeeper-5.5.4/lib/doorkeeper/orm/active_record.rb:35:in `block in initialize_application_owner!'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:71:in `class_eval'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:71:in `block in execute_hook'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:61:in `with_execution_control'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:66:in `execute_hook'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:52:in `block in run_load_hooks'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:51:in `each'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:51:in `run_load_hooks'
/bundle/vendor/ruby/3.0.0/gems/activerecord-6.1.6.1/lib/active_record/base.rb:315:in `<module:ActiveRecord>'
/bundle/vendor/ruby/3.0.0/gems/activerecord-6.1.6.1/lib/active_record/base.rb:15:in `<main>'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
/bundle/vendor/ruby/3.0.0/gems/doorkeeper-5.5.4/lib/doorkeeper/orm/active_record/application.rb:7:in `<module:Doorkeeper>'
/bundle/vendor/ruby/3.0.0/gems/doorkeeper-5.5.4/lib/doorkeeper/orm/active_record/application.rb:6:in `<main>'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
/usr/src/app/spec/factories/applications.rb:1:in `<main>'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:20:in `load'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:20:in `block (2 levels) in find_definitions'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:19:in `each'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:19:in `block in find_definitions'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:15:in `each'
/bundle/vendor/ruby/3.0.0/gems/factory_bot-6.2.1/lib/factory_bot/find_definitions.rb:15:in `find_definitions'
/bundle/vendor/ruby/3.0.0/gems/factory_bot_rails-6.2.0/lib/factory_bot_rails/railtie.rb:22:in `block in <class:Railtie>'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:68:in `block in execute_hook'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:61:in `with_execution_control'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:66:in `execute_hook'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:52:in `block in run_load_hooks'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:51:in `each'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/lazy_load_hooks.rb:51:in `run_load_hooks'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/application/finisher.rb:140:in `block in <module:Finisher>'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/initializable.rb:32:in `instance_exec'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/initializable.rb:32:in `run'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/initializable.rb:61:in `block in run_initializers'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/initializable.rb:60:in `run_initializers'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/application.rb:391:in `initialize!'
/usr/src/app/config/environment.rb:5:in `<main>'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/bundle/vendor/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/dependencies.rb:332:in `block in require'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/dependencies.rb:299:in `load_dependency'
/bundle/vendor/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/dependencies.rb:332:in `require'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/application.rb:367:in `require_environment!'
/bundle/vendor/ruby/3.0.0/gems/railties-6.1.6.1/lib/rails/application.rb:533:in `block in run_tasks_blocks'
/bundle/vendor/ruby/3.0.0/gems/airbrake-11.0.3/lib/airbrake/rake.rb:17:in `execute'
/bundle/vendor/ruby/3.0.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/usr/local/bin/bundle:23:in `load'
/usr/local/bin/bundle:23:in `<main>'
Tasks: TOP => db:test:prepare => db:load_config => environment
```

**Comment by nbulaj** [comment]
Now Doorkeeper uses `autoload` for loading ORM files. Does this issue still appears?

**Comment by gamecreature** [comment]
@nbulaj I just tested the main branch with the `Doorkeeper.set` hack removed and I can confirm this issue has been resolved.

**Comment by nbulaj** [comment]
Should be fixed in 5.6.4 as well, so closing now
